modules/agent/pkg/controllers/bundledeployment/controller.go

```diff
@@ -159,6 +172,28 @@ func shouldRedeploy(bd *fleet.BundleDeployment) bool {
 	return *bd.Status.SyncGeneration != bd.Spec.Options.ForceSyncGeneration
 }
 
+func (h *handler) cleanupOldAgent(bd *fleet.BundleDeployment, modified []fleet.ModifiedStatus) error {
```

Comment:
nit: we should probably rename "modified" to "modifiedStatuses" to avoid confusion with the below

Reply:
done